packages/x-components/src/x-modules/history-queries/components/__tests__/my-history.spec.ts

```diff
@@ -0,0 +1,135 @@
+  installNewXPlugin({ store, initialXModules: [historyQueriesXModule] }, localVue);
+  resetXHistoryQueriesStateWith(store, { historyQueries });
+
+  const wrapper = mount(
+    {
+      template,
+      components: {
+        MyHistory
+      }
+    },
+    {
+      localVue,
+      store
+    }
+  );
+  return {
+    wrapper: wrapper.findComponent(MyHistory),
+    search(query) {
+      wrapper.vm.$x.emit('UserAcceptedAQuery', query);
+      return wrapper.vm.$nextTick();
+    },
+    getListItems() {
+      return wrapper.findAll(getDataTestSelector('suggestion-item'));
+    },
+    findAllInWrapper(selector) {
+      return wrapper.findAll(getDataTestSelector(selector));
+    }
+  };
+}
+
+describe('testing MyHistory component', () => {
+  it('is an XComponent which has an XModule', () => {
+    const { wrapper } = renderMyHistory();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('historyQueries');
+  });
+
+  it('does not render the component if history is empty', () => {
+    const { wrapper } = renderMyHistory();
+    expect(wrapper.html()).toEqual('');
+  });
+
+  it('renders the list of searched queries', async () => {
+    const { search, getListItems } = renderMyHistory();
+
+    await search('lego');
+    const suggestionsWrappers = getListItems();
+    expect(suggestionsWrappers.wrappers).toHaveLength(1);
+    expect(suggestionsWrappers.at(0).text()).toEqual('lego ✕');
+  });
+
+  it('allows changing history query content and render the list of history queries', () => {
+    const historyQueries = createHistoryQueries(
+      'moura',
+      'calamares',
+      'rubia galega',
+      'pulpo',
+      'cachelos',
+      'navajas',
+      'croquetas',
+      'zamburiñas'
+    );
+
+    const { findAllInWrapper } = renderMyHistory({
+      template: `
+          <MyHistory>
+            <template #suggestion-content="suggestionContentScope">
+              <img src="./history-icon.svg" data-test="suggestion-history-icon"/>
+              <span :data-index="suggestionContentScope.index"
+                    data-test="suggestion-content-slot"
+                    v-html="suggestionContentScope.queryHTML"></span>
+            </template>
+            <template #suggestion-remove-content>
+              <img src="./remove-icon.svg" data-test="suggestion-remove-icon"/>
+            </template>
+          </MyHistory>
+        `,
+      historyQueries
+    });
+
+    const suggestionContentWrappers = findAllInWrapper('suggestion-history-icon');
+    const suggestionRemoveWrappers = findAllInWrapper('suggestion-remove-icon');
+    const suggestionContentSlotWrappers = findAllInWrapper('suggestion-content-slot');
+
+    expect(suggestionContentWrappers).toHaveLength(historyQueries.length);
+    expect(suggestionRemoveWrappers).toHaveLength(historyQueries.length);
+    expect(suggestionContentSlotWrappers).toHaveLength(historyQueries.length);
```

Comment:
By testing the slot scope I meant doing this, assering that each one of the custom elements in the slot is rendering the needed information. Note that I also took the oportunity to rename the wrappers variables.

```suggestion
    const suggestionIconWrappers = findAllInWrapper('suggestion-history-icon');
    const suggestionRemoveIconWrappers = findAllInWrapper('suggestion-remove-icon');
    const suggestionContentWrappers = findAllInWrapper('suggestion-content-slot');

    expect(suggestionIconWrappers).toHaveLength(historyQueries.length);
    expect(suggestionRemoveIconWrappers).toHaveLength(historyQueries.length);
    expect(suggestionContentWrappers).toHaveLength(historyQueries.length);
    suggestionContentWrappers.wrappers.forEach((contentWrapper, index) => {
      expect(contentWrapper.attributes('data-index')).toEqual(index.toString());
      expect(contentWrapper.text()).toEqual(historyQueries[index].query);
    });
```